modules/product/src/main/java/com/opengamma/strata/product/Security.java

```diff
@@ -1,62 +1,74 @@
+ * Within Strata, a financial instrument is a security if it is a standardized
+ * exchange-based contract that can be referenced by a {@link SecurityId}.
+ * The security can be looked up in {@link ReferenceData} using the identifier.
+ * <p>
  * It is intended that Over-The-Counter (OTC) instruments, such as an interest rate swap,
  * are embedded directly within the trade, rather than handled as one-off securities.
  * <p>
- * When referring to a security from another object, such as an underlying on a
- * more complex trade, consideration should be given to using {@link SecurityLink}.
- * <p>
  * Implementations of this interface must be immutable beans.
- * 
- * @param <P>  the type of the product
  */
-public interface Security<P extends Product>
-    extends IdentifiableBean {
+public interface Security {
 
   /**
-   * The primary standard identifier for the security.
+   * Gets the security identifier.
    * <p>
-   * The standard identifier is used to identify the security.
-   * It will typically be an identifier in an external data system.
-   * <p>
-   * A security may have multiple active identifiers. Any identifier may be chosen here.
-   * Certain uses of the identifier, such as storage in a database, require that the
-   * identifier does not change over time, and this should be considered best practice.
+   * This identifier uniquely identifies the security within the system.
+   * 
+   * @return the security identifier
    */
-  @Override
-  public abstract StandardId getStandardId();
+  public abstract SecurityInfo getSecurityInfo();
 
   /**
-   * Gets the entire set of additional attributes.
+   * Gets the security identifier.
    * <p>
-   * Attributes are typically used to tag the object with additional information.
+   * This identifier uniquely identifies the security within the system.
+   * 
+   * @return the security identifier
+   */
+  public default SecurityId getSecurityId() {
+    return getSecurityInfo().getId();
+  }
+
+  /**
+   * Gets the currency that the security is traded in.
    * 
-   * @return the complete set of attributes
+   * @return the trading currency
    */
-  public abstract ImmutableMap<String, String> getAttributes();
+  public default Currency getCurrency() {
+    return getSecurityInfo().getPriceInfo().getCurrency();
+  }
 
+  //-------------------------------------------------------------------------
   /**
-   * Gets the product underlying the security.
+   * Creates a trade based on this security.
    * <p>
-   * All securities refer to a single underlying product.
-   * The product captures the financial details of the security contract.
+   * This creates a trade of a suitable type for this security.
    * 
-   * @return the product
+   * @param tradeInfo  the trade information
+   * @param quantity  the number of contracts in the trade
+   * @param tradePrice  the price that was traded, in decimal form
```

Comment:
Isn't "in decimal form" redundant? What other form could it be given it's a `double`?